backvendor/gosource.go

```diff
@@ -187,52 +187,29 @@ func findImportComment(src *GoSource) (string, error) {
 		if err != nil {
 			return err
 		}
-		if info.IsDir() {
-			if info.Name() != "." &&
-				strings.HasPrefix(info.Name(), ".") {
-				return filepath.SkipDir
-			}
-			if info.Name() == "vendor" || info.Name() == "testdata" {
-				return filepath.SkipDir
-			}
+		if !info.IsDir() {
 			return nil
 		}
-		if !strings.HasSuffix(info.Name(), ".go") {
-			return nil
+		if info.Name() != "." &&
+			strings.HasPrefix(info.Name(), ".") {
+			return filepath.SkipDir
 		}
-		r, err := os.Open(pth)
-		if err != nil {
-			return err
+		switch info.Name() {
+		case "vendor", "testdata":
```

Comment:
This makes sense, but adding a comment would be appreciated. Something along the lines of:
```go
// Skip these special directories since "vendor" contains local copies of dependencies and "testdata" includes data files only used for testing that can be safely ignored.
```